include/nuttx/fs/fs.h

```diff
@@ -215,20 +217,26 @@ struct file_operations
    */
 
   CODE int     (*close)(FAR struct file *filep);
-  CODE ssize_t (*read)(FAR struct file *filep, FAR char *buffer,
-                       size_t buflen);
-  CODE ssize_t (*write)(FAR struct file *filep, FAR const char *buffer,
-                        size_t buflen);
+  CODE ssize_t (*readv)(FAR struct file *filep, FAR const struct iovec *iov,
+                        int iovcnt);
+  CODE ssize_t (*writev)(FAR struct file *filep, FAR const struct iovec *iov,
+                         int iovcnt);
   CODE off_t   (*seek)(FAR struct file *filep, off_t offset, int whence);
   CODE int     (*ioctl)(FAR struct file *filep, int cmd, unsigned long arg);
   CODE int     (*mmap)(FAR struct file *filep,
                        FAR struct mm_map_entry_s *map);
   CODE int     (*truncate)(FAR struct file *filep, off_t length);
 
-  /* The two structures need not be common after this point */
-
   CODE int     (*poll)(FAR struct file *filep, FAR struct pollfd *fds,
                        bool setup);
+  CODE ssize_t (*compat_read)(FAR struct file *filep, FAR char *buffer,
```

Comment:
could we keep read/write callback but add readv/writev to avoid changing so many files?

Reply:
i structured this way because i think we should retire read/write ops after migrating everything to iovec-based.

Reply:
you can remove read/write from the whole code base after all drivers/fs change to readv/writev, it doesn't make sense to change read/write to compact_read/compact_write and then remove it later.